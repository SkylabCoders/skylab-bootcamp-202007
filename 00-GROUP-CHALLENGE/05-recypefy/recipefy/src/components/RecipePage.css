.time__background {
  background: url("https://image.flaticon.com/icons/svg/2972/2972562.svg");
  background-position: center center;
  background-size: 75%;
  background-repeat: no-repeat;
}

p {
  color: black;
}

.image-recipe {
  min-width: 250px;
  margin-top: 10px;
}

.calories__background {
  background: url("https://image.flaticon.com/icons/svg/2738/2738728.svg");
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 60%;
}

.yield__background {
  background: url("https://image.flaticon.com/icons/svg/1021/1021444.svg");
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 60%;
}

.flex__row {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
}

.flex__column {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  text-align: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.rate__recipe {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.info__container {
  margin-top: 70px;
  width: 100%;
  -ms-flex-pack: distribute;
      justify-content: space-around;
}

.buttons__container {
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  width: 100%;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  margin-top: 20px;
}

.ingredients__container {
  text-align: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  margin-top: 10%;
  margin-bottom: 10%;
}

.data__container {
  height: 100%;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
}

.data__circle {
  width: 80px;
  height: 80px;
  border-radius: 50%;
  padding: 3px;
  text-align: center;
}

.title__img__container {
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.title__img__container img {
  margin-top: 15px;
  width: 40vw;
  max-width: 250px;
  border-radius: 15px;
}

.error_text {
  text-align: center;
}

.underphoto__button {
  padding: 15px;
  width: 20vw;
  background-color: #f8f8f0;
  border-radius: 10px;
  margin-bottom: 20px;
}

a {
  text-align: center;
}

.rate__elem {
  margin-right: 10px;
}

.desktop__recipe__title {
  display: none;
}

.ingredients__title {
  margin: 10px;
  font-size: 25px;
}

.ingredients__text {
  margin: 10px;
  text-align: start;
  max-width: 300px;
}

.ingredients__source {
  margin-top: 20px;
  color: #373783;
  border: solid 2px #373783;
  padding: 5px;
  border-radius: 10px;
}

.text__icon {
  margin-bottom: 5px;
  text-align: center;
  font-size: 16px;
}

.box__icon {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.recipe__icon-list {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  margin-bottom: 10px;
}

.desktop__display--left {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.desktop__display--right {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.underphoto__button--source {
  margin-top: 20px;
  background-color: #65dfc3;
  padding: 15px;
  border-radius: 10px;
}

@media screen and (min-width: 600px) {
  .underphoto__button {
    margin-top: 10vh;
  }
}

@media screen and (min-width: 1024px) {
  .desktop__display {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
  }
  .desktop__display--left {
    -webkit-box-ordinal-group: 1;
        -ms-flex-order: 0;
            order: 0;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    width: 50vw;
    height: 100%;
    margin-top: 80px;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
  }
  .desktop__display--right {
    -webkit-box-ordinal-group: 2;
        -ms-flex-order: 1;
            order: 1;
    width: 50vw;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
  }
  .buttons__container {
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
    margin-top: 49px;
    margin-bottom: 20px;
    height: 100%;
    width: 100%;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
  }
  .data__container {
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    margin-bottom: 10px;
    margin-top: 20px;
  }
  .info__container {
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    margin-top: 110px;
    width: 100%;
    -ms-flex-pack: distribute;
        justify-content: space-around;
  }
  .info__container h2 {
    display: none;
  }
  .info__container img {
    max-width: 350px;
  }
  .ingredients__container {
    height: 40vh;
  }
  .underphoto__button {
    margin-bottom: 10px;
    margin-top: 5vh;
  }
  .preference__box {
    height: 80px;
    margin-left: 0px;
  }
  .desktop__recipe__title {
    display: inline-block;
    text-align: center;
    margin-bottom: 15px;
  }
  .ingredients__text {
    max-width: 500px;
  }
}
/*# sourceMappingURL=RecipePage.css.map */