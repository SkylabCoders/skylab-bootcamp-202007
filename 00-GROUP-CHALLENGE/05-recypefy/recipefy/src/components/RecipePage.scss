.time__background {
	background: url('https://image.flaticon.com/icons/svg/2972/2972562.svg');
	background-position: center center;
	background-size: 75%;
	background-repeat: no-repeat;
}

p {
	color: black;
}


.image-recipe {
	min-width: 250px;
	margin-top: 10px;
}

.calories__background {
	background: url('https://image.flaticon.com/icons/svg/2738/2738728.svg');
	background-position: center center;
	background-repeat: no-repeat;
	background-size: 60%;
}

.yield__background {
	background: url('https://image.flaticon.com/icons/svg/1021/1021444.svg');
	background-position: center center;
	background-repeat: no-repeat;
	background-size: 60%;
}

.flex__row {
	display: flex;
	flex-direction: row;
}

.flex__column {
	display: flex;
	flex-direction: column;
	text-align: center;
	justify-content: center;
	align-items: center;
}

.rate__recipe {
	align-items: center;
	justify-content: center;
}


.info__container {
	margin-top: 70px;
	width: 100%;
	justify-content: space-around;
}

.buttons__container {
	justify-content: center;
	width: 100%;
	align-items: center;
	display: flex;
	flex-direction: column;
	margin-top: 20px;
}

.ingredients__container {
	text-align: center;
	align-items: center;
	height: fit-content;
	margin-top: 10%;
	margin-bottom: 10%;
}

.data__container {
	height: 100%;
	justify-content: space-evenly;
}

.data__circle {
	width: 80px;
	height: 80px;

	border-radius: 50%;
	padding: 3px;
	text-align: center;
}

.title__img__container {
	justify-content: center;
	align-items: center;

	img {
		margin-top: 15px;
		width: 40vw;
		max-width: 250px;
		border-radius: 15px;
	}

}

.error_text {
	text-align: center;
}

.underphoto__button {
	padding: 15px;
	width: 20vw;
	background-color: #f8f8f0;
	border-radius: 10px;
	margin-bottom: 20px;
}

a {
	text-align: center;
}

.rate__elem {
	margin-right: 10px;
}

.desktop__recipe__title {
	display: none;
}

.ingredients__title {
	margin: 10px;
	font-size: 25px;
}

.ingredients__text {
	margin: 10px;
	text-align: start;
	max-width: 300px;
}

.ingredients__source {
	margin-top: 20px;
	color: rgb(55, 55, 131);
	border: solid 2px rgb(55, 55, 131);
	padding: 5px;
	border-radius: 10px;
}

.text__icon {
	margin-bottom: 5px;
	text-align: center;
	font-size: 16px;
}

.box__icon {
	display: flex;
	flex-direction: column;
	justify-content: center;
}

.recipe__icon-list {
	display: flex;
	flex-direction: row;
	justify-content: center;
	margin-bottom: 10px;
}

.desktop__display--left {
	display: flex;
	flex-direction: column;
	align-items: center;
}

.desktop__display--right {
	display: flex;
	flex-direction: column;
	align-items: center;

}

.underphoto__button--source {
	margin-top: 20px;
	background-color: #65dfc3;
	padding: 15px;
	border-radius: 10px;
}

@media screen and (min-width: 600px) {
	.underphoto__button {
		margin-top: 10vh;
	}

}

@media screen and (min-width: 1024px) {
	.desktop__display {
		display: flex;
		flex-direction: row;
	}



	.desktop__display--left {
		order: 0;
		display: flex;
		flex-direction: column;
		width: 50vw;
		height: 100%;
		margin-top: 80px;
		align-items: center;
	}

	.desktop__display--right {
		order: 1;
		width: 50vw;
		align-items: center;
	}

	.buttons__container {
		justify-content: center;
		margin-top: 49px;
		margin-bottom: 20px;
		height: 100%;
		width: 100%;
		flex-direction: column;
		align-items: center;
	}

	.data__container {
		flex-direction: column;
		margin-bottom: 10px;
		margin-top: 20px;
	}

	.info__container {
		flex-direction: column;
		margin-top: 110px;
		width: 100%;
		justify-content: space-around;

		h2 {
			display: none;
		}

		img {
			max-width: 350px;
		}
	}

	.ingredients__container {
		height: 40vh;
	}

	.underphoto__button {
		margin-bottom: 10px;
		margin-top: 5vh;
	}

	.preference__box {
		height: 80px;
		margin-left: 0px;
	}

	.desktop__recipe__title {
		display: inline-block;
		text-align: center;
		margin-bottom: 15px;
	}

	.ingredients__text {
		max-width: 500px;
	}
}