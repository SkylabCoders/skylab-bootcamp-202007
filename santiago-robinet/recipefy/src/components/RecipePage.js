import React from 'react';


function RecipePage(){
    return(
        <>
          <p> I'm da RecipePage!! And I work!!</p>
        </>
    );
}

export default RecipePage;