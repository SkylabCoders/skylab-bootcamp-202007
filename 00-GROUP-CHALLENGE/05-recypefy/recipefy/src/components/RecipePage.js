import React, { useState, useEffect } from 'react';
import receteListMoked from '../Recipe.mok.js';
import './RecipePage.css';
import recipe from '../stores/RecipeStore';
import { NavLink } from 'react-router-dom';

function RecipePage() {

	let balanced = false;
	let protein = false;
	let low_fat = false;
	let low_carb = false;
	let vegan = false;
	let vegetarian = false;
	let sugar = false;
	let peanut = false;
	let treenut = false;
	let alcohol = false;


	let [recipeElement] = useState(recipe.getRecipeByTitle(getUrl()));
	const [titleString] = useState(recipeElement.title);
	let [time] = useState(`${recipeElement.time} min`);
	if (time === '0 min') {
		time = '65 min';
	}
	const [calories] = useState(recipeElement.calories.toFixed(2));
	let [yeld] = useState(recipeElement.yeld);
	if (yeld === null || yeld === undefined || yeld === '') {
		yeld = 4;
	}

	let [ingredients] = useState(recipeElement.ingredients);

	const [photo] = useState(recipeElement.photo);
	const [url] = useState(recipeElement.url);
	const [source] = useState(`source: ${recipeElement.source}`);

	function getUrl() {
		//Se obtiene el valor de la URL desde el navegador
		var actual = window.location + '';
		//Se realiza la divisi√≥n de la URL
		var split = actual.split('/');
		//Se obtiene el ultimo valor de la URL
		var title = split[split.length - 1];

		return decodeURI(title);
	}

	for (let i = 0; i < recipeElement.preferences.length - 1; i++) {
		if (recipeElement.preferences[i] === 'Balanced') {
			balanced = true;
		} else if (recipeElement.preferences[i] === 'High-Protein') {
			protein = true;
		} else if (recipeElement.preferences[i] === 'Low-Fat') {
			low_fat = true;
		} else if (recipeElement.preferences[i] === 'Low-Carb') {
			low_carb = true;
		} else if (recipeElement.preferences[i] === 'Vegan') {
			vegan = true;
		} else if (recipeElement.preferences[i] === 'Vegetarian') {
			vegetarian = true;
		} else if (recipeElement.preferences[i] === 'Sugar-Conscious') {
			sugar = true;
		} else if (recipeElement.preferences[i] === 'Peanut-Free') {
			peanut = true;
		} else if (recipeElement.preferences[i] === 'Tree-Nut-Free') {
			treenut = true;
		} else if (recipeElement.preferences[i] === 'Alcohol-Free') {
			alcohol = true;
		}
	}

	debugger;

	return (
		<div className="desktop__display">
			<div className="desktop__display--right">
				<section className="flex__row info__container">
					<div className="flex__column data__container">
						<div className="recipe__icon-list">
							<div className="box__icon">
								<p className="text__icon">{time}</p>
								<div id="time" className="data__circle time__background">
								</div>
							</div>
							<div className="box__icon">
								<p className="text__icon">{calories}</p>
								<div id="calories" className="data__circle calories__background">
								</div>
							</div>
							<div className="box__icon">
								<p className="text__icon">{yeld}</p>
								<div id="yeld" className="data__circle yield__background">
								</div>
							</div>
						</div>
						<div className="title__img__container flex__column">
							<h2>{titleString}</h2>
							<img id="image-recipe" src={photo} />
							<p className="ingredients__source">{source}</p>
						</div>
					</div>
				</section>
			</div>

			<div className="desktop__display--left">
				<section>
					<div className="flex__row buttons__container">
						<div className="flex__column">
							<h2 id="title" className="desktop__recipe__title">
								{titleString}
							</h2>

							<div className="underphoto__button preference__box">
								<div className="recipe__text--preferences">
									{balanced && (
										<img
											className="preferences__icon preference__balanced"
											id="balanced"
											src="https://image.flaticon.com/icons/png/512/30/30636.png"
										></img>
									)}
									{protein && (
										<img
											className="preferences__icon"
											id="protein"
											src="https://icon-library.com/images/protein-icon/protein-icon-0.jpg"
										></img>
									)}
									{low_fat && (
										<img
											className="preferences__icon"
											id="low_fat"
											src="https://www.pinclipart.com/picdir/big/150-1505070_low-fat-or-low-carb-icon-clipart.png"
										></img>
									)}
									{low_carb && (
										<img
											className="preferences__icon"
											id="low_carb"
											src="https://cdn2.iconfinder.com/data/icons/organic-food-1/24/Low_Carb-512.png"
										></img>
									)}
									{vegan && (
										<img
											className="preferences__icon"
											id="vegan"
											src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/5b/Vegan_friendly_icon.svg/900px-Vegan_friendly_icon.svg.png"
										></img>
									)}

									{vegetarian && (
										<img
											className="preferences__icon"
											id="vegetarian"
											src="https://img.icons8.com/plasticine/100/000000/vegetarian-food.png"
										></img>
									)}
									{sugar && (
										<img
											className="preferences__icon"
											id="sugar"
											src="https://www.footys.co.za/images/icon-sugar.png"
										></img>
									)}
									{peanut && (
										<img
											className="preferences__icon preference__peanut-free"
											id="peanut"
											src="https://cdn0.iconfinder.com/data/icons/food-product-labels/128/peanut-free-512.png"
										></img>
									)}
									{treenut && (
										<img
											className="preferences__icon preference__tree-nut-free"
											id="treenut"
											src="https://cdn3.iconfinder.com/data/icons/food-allergens-3/77/allergens-tree-nut-free-512.png"
										></img>
									)}
									{alcohol && (
										<img
											className="preferences__icon preference__alcohol-free"
											id="alcohol"
											src="https://cdn3.iconfinder.com/data/icons/food-allergens-3/77/allergens-alcohol-free-512.png"
										></img>
									)}
								</div>
							</div>
						</div>
						<a href={url} className="underphoto__button">
							LET'S COOK IT!<br></br> CLICK HERE
						</a>
					</div>
				</section>
				<section className="flex__column ingredients__container">
					<p className="ingredients__title">INGREDIENTS</p>
					<ul className="ingredients__text" id="ingredients">
						{ingredients.map((ingredient) => (
							<li>{ingredient}</li>
						))}
					</ul>

				</section>
			</div>
		</div>
	);
}

export default RecipePage;
