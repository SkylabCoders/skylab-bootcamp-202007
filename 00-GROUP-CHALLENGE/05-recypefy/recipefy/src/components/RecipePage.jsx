import React, { useState, useEffect } from 'react';
import './RecipePage.scss';
import recipe from '../stores/RecipeStore';
import recipeStore from '../stores/RecipeStore';
function RecipePage() {
	const [balanced, setBalanced] = useState(false);
	const [protein, setProtein] = useState(false);
	const [lowFat, setLowFat] = useState(false);
	const [lowCarb, setLowCarb] = useState(false);
	const [vegan, setVegan] = useState(false);
	const [vegetarian, setVegetarian] = useState(false);
	const [sugar, setSugar] = useState(false);
	const [peanut, setPeanut] = useState(false);
	const [treenut, setTreenut] = useState(false);
	const [alcohol, setAlcohol] = useState(false);
	const [recipeElement, setRecipeElement] = useState(
		recipe.getRecipeByTitle(getUrl())
	);
	const [titleString] = useState(getUrl());
	const [time, setTime] = useState('');
	const [calories, setCalories] = useState('');
	const [yeld, setYeld] = useState('');
	const [ingredients, setIngredients] = useState([]);
	const [photo, setPhoto] = useState('');
	const [url, setUrl] = useState('');
	const [sourceItem, setSourceItem] = useState(``);
	let [preferencesList, setPreferencesList] = useState();

	useEffect(() => {
		recipeStore.addChangeListener(onChange);
		if (!recipeElement) {
			setRecipeElement(recipe.getRecipeByTitle(getUrl()));
		} else {
			{
				if (!recipeElement.time) {
					setTime('--');
				} else {
					setTime(`${recipeElement.time} min`);
				}
				setCalories(recipeElement.calories.toFixed(2));
				if (yeld === null || yeld === undefined || yeld === '') {
					setYeld('--');
				} else {
					setYeld(recipeElement.yeld);
				}
				setIngredients(recipeElement.ingredients);
				setPhoto(recipeElement.photo);
				setUrl(recipeElement.url);
				setSourceItem(`source: ${recipeElement.source}`);
				setPreferencesList(recipeElement.preferences);
				if (preferencesList) {
					changePreferences(preferencesList);
				}
			}
		}
		return () => recipeStore.removeChangeListener(onChange);
	}, [recipeElement, preferencesList]);

	function changePreferences(listOfPreferences) {
		for (let i = 0; i < listOfPreferences.length; i++) {
			if (listOfPreferences[i] === 'Balanced') {
				setBalanced(true);
			} else if (listOfPreferences[i] === 'High-Protein') {
				setProtein(true);
			} else if (listOfPreferences[i] === 'Low-Fat') {
				setLowFat(true);
			} else if (listOfPreferences[i] === 'Low-Carb') {
				setLowCarb(true);
			} else if (listOfPreferences[i] === 'Vegan') {
				setVegan(true);
			} else if (listOfPreferences[i] === 'Vegetarian') {
				setVegetarian(true);
			} else if (listOfPreferences[i] === 'Sugar-Conscious') {
				setSugar(true);
			} else if (listOfPreferences[i] === 'Peanut-Free') {
				setPeanut(true);
			} else if (listOfPreferences[i] === 'Tree-Nut-Free') {
				setTreenut(true);
			} else if (listOfPreferences[i] === 'Alcohol-Free') {
				setAlcohol(true);
			}
		}
	}
	function onChange() {
		setRecipeElement(recipe.getRecipeByTitle(getUrl()));
	}

	function getUrl() {
		var actual = window.location + '';
		var split = actual.split('/');
		var title = split[split.length - 1];

		return decodeURI(title);
	}

	return (
		<div className="desktop__display body-content">
			<div className="desktop__display--right">
				<section className="flex__row info__container">
					<div className="flex__column data__container">
						<div className="recipe__icon-list">
							<div className="box__icon">
								<p className="text__icon">{time}</p>
								<div id="time" className="data__circle time__background"></div>
							</div>
							<div className="box__icon">
								<p className="text__icon">{calories}</p>
								<div
									id="calories"
									className="data__circle calories__background"
								></div>
							</div>
							<div className="box__icon">
								<p className="text__icon">{yeld}</p>
								<div id="yeld" className="data__circle yield__background"></div>
							</div>
						</div>
						<div className="title__img__container flex__column">
							<h2>{titleString}</h2>
							<img
								alt=""
								className="image-recipe"
								id="image-recipe"
								src={photo}
							/>
							<p className="ingredients__source">{sourceItem}</p>
						</div>
					</div>
				</section>
			</div>

			<div className="desktop__display--left">
				<section>
					<div className="flex__row buttons__container">
						<div className="flex__column">
							<h2 id="title" className="desktop__recipe__title">
								{titleString}
							</h2>

							<div className="underphoto__button">
								<div className="recipe__text--preferences">
									{balanced && (
										<img
											alt=""
											className="preferences__icon preference__balanced"
											id="balanced"
											src="https://image.flaticon.com/icons/png/512/30/30636.png"
										></img>
									)}
									{protein && (
										<img
											alt=""
											className="preferences__icon"
											id="protein"
											src="https://icon-library.com/images/protein-icon/protein-icon-0.jpg"
										></img>
									)}
									{lowFat && (
										<img
											alt=""
											className="preferences__icon"
											id="low_fat"
											src="https://www.pinclipart.com/picdir/big/150-1505070_low-fat-or-low-carb-icon-clipart.png"
										></img>
									)}
									{lowCarb && (
										<img
											alt=""
											className="preferences__icon"
											id="low_carb"
											src="https://cdn2.iconfinder.com/data/icons/organic-food-1/24/Low_Carb-512.png"
										></img>
									)}
									{vegan && (
										<img
											alt=""
											className="preferences__icon"
											id="vegan"
											src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/5b/Vegan_friendly_icon.svg/900px-Vegan_friendly_icon.svg.png"
										></img>
									)}

									{vegetarian && (
										<img
											alt=""
											className="preferences__icon"
											id="vegetarian"
											src="https://img.icons8.com/plasticine/100/000000/vegetarian-food.png"
										></img>
									)}
									{sugar && (
										<img
											alt=""
											className="preferences__icon"
											id="sugar"
											src="https://www.footys.co.za/images/icon-sugar.png"
										></img>
									)}
									{peanut && (
										<img
											alt=""
											className="preferences__icon preference__peanut-free"
											id="peanut"
											src="https://cdn0.iconfinder.com/data/icons/food-product-labels/128/peanut-free-512.png"
										></img>
									)}
									{treenut && (
										<img
											alt=""
											className="preferences__icon preference__tree-nut-free"
											id="treenut"
											src="https://cdn3.iconfinder.com/data/icons/food-allergens-3/77/allergens-tree-nut-free-512.png"
										></img>
									)}
									{alcohol && (
										<img
											alt=""
											className="preferences__icon preference__alcohol-free"
											id="alcohol"
											src="https://cdn3.iconfinder.com/data/icons/food-allergens-3/77/allergens-alcohol-free-512.png"
										></img>
									)}
								</div>
								{!preferencesList && (
									<p className="error_text">No preferences</p>
								)}
							</div>
						</div>
						<a href={url} className="underphoto__button--source">
							LET'S COOK IT!<br></br> CLICK HERE
						</a>
					</div>
				</section>
				<section className="flex__column ingredients__container">
					<p className="ingredients__title">INGREDIENTS</p>
					<ul className="ingredients__text" id="ingredients">
						{ingredients.map((ingredient) => (
							<li>{ingredient}</li>
						))}
					</ul>
				</section>
			</div>
		</div>
	);
}

export default RecipePage;
