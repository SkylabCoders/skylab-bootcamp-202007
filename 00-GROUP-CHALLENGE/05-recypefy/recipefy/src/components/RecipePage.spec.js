import React from 'react';
import RecipePage from './RecipePage';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

function renderRecipePage(arg) {
	const defaultProps = {
		match: {
			params: {}
		}
	};
	const props = { ...defaultProps, ...arg };
	return renderer.create(
		<BrowserRouter>
			<RecipePage {...props} />
		</BrowserRouter>
	);
}

describe('RecipePage', () => {
	const title = 'Grilled Tofu Salad With Miso Dressing';
	let recipePageTree;

	beforeEach(async () => {
		recipePageTree = renderRecipePage();
		recipePageTree.update();
	});

	fit('should create recipe page without a recipe', async () => {
		const props = {
			match: {
				params: {
					recipeId: title
				}
			}
		};
		recipePageTree = renderRecipePage(props);
		expect(recipePageTree).toMatchSnapshot();
	});
});
