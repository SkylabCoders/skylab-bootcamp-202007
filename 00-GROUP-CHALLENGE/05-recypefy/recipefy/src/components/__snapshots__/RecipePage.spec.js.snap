// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage should create recipe page without a recipe 1`] = `
<div
  className="desktop__display body-content"
>
  <div
    className="desktop__display--right"
  >
    <section
      className="flex__row info__container"
    >
      <div
        className="flex__column data__container"
      >
        <div
          className="recipe__icon-list"
        >
          <div
            className="box__icon"
          >
            <p
              className="text__icon"
            >
              
            </p>
            <div
              className="data__circle time__background"
              id="time"
            />
          </div>
          <div
            className="box__icon"
          >
            <p
              className="text__icon"
            >
              
            </p>
            <div
              className="data__circle calories__background"
              id="calories"
            />
          </div>
          <div
            className="box__icon"
          >
            <p
              className="text__icon"
            >
              
            </p>
            <div
              className="data__circle yield__background"
              id="yeld"
            />
          </div>
        </div>
        <div
          className="title__img__container flex__column"
        >
          <h2>
            
          </h2>
          <img
            alt=""
            className="image-recipe"
            id="image-recipe"
            src=""
          />
          <p
            className="ingredients__source"
          >
            
          </p>
        </div>
      </div>
    </section>
  </div>
  <div
    className="desktop__display--left"
  >
    <section>
      <div
        className="flex__row buttons__container"
      >
        <div
          className="flex__column"
        >
          <h2
            className="desktop__recipe__title"
            id="title"
          >
            
          </h2>
          <div
            className="underphoto__button"
          >
            <div
              className="recipe__text--preferences"
            />
            <p
              className="error_text"
            >
              No preferences
            </p>
          </div>
        </div>
        <a
          className="underphoto__button--source"
          href=""
        >
          LET'S COOK IT!
          <br />
           CLICK HERE
        </a>
      </div>
    </section>
    <section
      className="flex__column ingredients__container"
    >
      <p
        className="ingredients__title"
      >
        INGREDIENTS
      </p>
      <ul
        className="ingredients__text"
        id="ingredients"
      />
    </section>
  </div>
</div>
`;
